torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -96,6 +96,23 @@ static int64_t floordiv(int64_t a, int64_t b) {
     return (r.rem) ? r.quot - 1 : r.quot;
   }
 }
+void checkDoubleInRange(double a) {
+  if (std::isnan(a) || std::isinf(a) ||
+      a > double(std::numeric_limits<int64_t>::max()) ||
+      a < double(std::numeric_limits<int64_t>::min())) {
+    throw c10::Error(
```

Comment:
Do we have a test to trigger this error btw?

Reply:
Not now. I don't think it's too necessary - our behavior when the user passes in an invalid input is 1. not really defined, and 2. more permissive than Python (other than with >64 bit integers). I think it's likely that any tests we write are likely to become code churn as various things change.